definitions/mongodb.rb

```diff
@@ -20,27 +20,23 @@
 #
 
 define :mongodb_instance,
-       mongodb_type: 'mongod',
        action: [:enable, :start],
        logpath: '/var/log/mongodb/mongod.log',
        configservers: [],
        replicaset: nil,
+       is_mongos: nil,
        notifies: [] do
-  # TODO: this is the only remain use of params[:mongodb_type], is it still needed?
-  unless %w(mongod configserver mongos).include?(params[:mongodb_type])
```

Comment:
Aren't you removing `configserver` from configuration?

And also, couldn't you interpret `is_shard` as being `is_mongos`?
I don't understand much about mongo itself just checking if my assumption is right

Reply:
I don't see `mongodb_type = "configserver"` value being used, just the `is_configserver`. Does that seem sane?

As for mongos vs is_shard, I honestly don't remember. 

Looks like `is_shard` only triggers this while in `mongod` mode (non `is_mongos`):
```
      new_resource.config['sharding'] ||= {}
      new_resource.config['sharding']['clusterRole'] = 'shardsvr'
```

It also affects the default `replicaset_name` and (as a node attribute) is used for cluster discovery.

Honestly, the names of some of these things are very confusing, any suggestions would be appreciated.